File: GenerateRendererSample/objective-c/GenerateRendererSample/Controllers/OptionsViewController.m
/*
 Copyright 2014 Esri
 
 Licensed under the Apache License, Version 2.0 (the "License");
 you may not use this file except in compliance with the License.
 You may obtain a copy of the License at
 
 http://www.apache.org/licenses/LICENSE-2.0
 
 Unless required by applicable law or agreed to in writing, software
 distributed under the License is distributed on an "AS IS" BASIS,
 WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License.
 */

#import "OptionsViewController.h"

@interface OptionsViewController ()

@end

@implementation OptionsViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    //clear table background color
    [self.tableView setBackgroundColor:[UIColor clearColor]];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

//setter method for options
//reloading tableview everytime options is assigned a value
-(void)setOptions:(NSArray *)options {
    _options = options;
    
    [self.tableView reloadData];
}

#pragma mark - Table view data source

- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView {
    // Return the number of sections.
    return 1;
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    // Return the number of rows in the section.
    return self.options.count;
}

-(UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    NSString *reuseIdentifier = @"OptionCell";
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:reuseIdentifier];
    
    if (!cell) {
        cell = [[UITableViewCell alloc] initWithStyle:UITableViewCellStyleDefault reuseIdentifier:reuseIdentifier];
    }
    NSString *option = [self.options objectAtIndex:indexPath.row];
    cell.textLabel.text = option;
    //clear cell background color
    [cell setBackgroundColor:[UIColor clearColor]];
    return cell;
}

-(void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath {
    //notify the delegate
    if (self.delegate && [self.delegate respondsToSelector:@selector(optionsViewController:didSelectIndex:forTextField:)]) {
        [self.delegate optionsViewController:self didSelectIndex:indexPath.row forTextField:self.textField];
    }
}

@end

File: GenerateRendererSample/swift/GenerateRenderer/Controllers/OptionsViewController.swift
// Copyright 2014 ESRI
//
// All rights reserved under the copyright laws of the United States
// and applicable international laws, treaties, and conventions.
//
// You may freely redistribute and use this sample code, with or
// without modification, provided you include the original copyright
// notice and use restrictions.
//
// See the use restrictions at http://help.arcgis.com/en/sdk/10.0/usageRestrictions.htm

import UIKit

protocol OptionsViewControllerDelegate:class  {
    
    //delegate used to notify that an option was selected
    func optionsViewController(optionsViewController: OptionsViewController, didSelectIndex index:(NSInteger), forTextField textField:(UITextField))
}

class OptionsViewController: UITableViewController {

    //array to store the list of options as NSString
    var options:[String]! {
        didSet {
            self.tableView.reloadData()
        }
    }
    //UITextField for which the options need to be displayed
    var textField:UITextField!
    
    weak var delegate:OptionsViewControllerDelegate?
    
    override func viewDidLoad() {
        super.viewDidLoad()

        //clear table background color
        self.tableView.backgroundColor = UIColor.clearColor()
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    //MARK: - table view datasource
    
    override func numberOfSectionsInTableView(tableView: UITableView) -> Int {
        return 1
    }

    override func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        if self.options != nil {
            return self.options.count
        }
        return 0
    }
    
    override func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
        let reuseIdentifier = "OptionCell"
        var cell = tableView.dequeueReusableCellWithIdentifier(reuseIdentifier)
        
        if cell == nil {
            cell = UITableViewCell(style: .Default, reuseIdentifier: reuseIdentifier)
        }
        let option = self.options[indexPath.row]
        cell!.textLabel!.text = option
        //clear cell background color
        cell!.backgroundColor = UIColor.clearColor()
        return cell!
    }
    
    //MARK: - tableview delegate
    
    override func tableView(tableView: UITableView, didSelectRowAtIndexPath indexPath: NSIndexPath) {
        //notify the delegate
        self.delegate?.optionsViewController(self, didSelectIndex:indexPath.row, forTextField:self.textField)
    }
}
